cmd/kubeadm/app/cmd/join.go

```diff
@@ -268,6 +268,18 @@ func (j *Join) Run(out io.Writer) error {
 		return fmt.Errorf("couldn't save the CA certificate to disk: %v", err)
 	}
 
+	// ReadSentinelFile check sentinelfile exist
+	data, err := kubeadmutil.ReadSentinelFile()
+	if err != nil {
+		// WriteSentinelFile is record kubeadmin join to sentinelfile
+		if err := kubeadmutil.WriteSentinelFile("kubeadm join"); err != nil {
```

Comment:
We need the whole command line string in order for this to be useful imo.

Reply:
Which should probably be at the base of all commands.

Reply:
i've added the full command for the plan for phase 2:
https://github.com/kubernetes/kubeadm/issues/997

this current PR is what we've discussed with @chuckha about phase 1 - being like the minimum sentinel check. but i guess it can go for phase 1 as well.